201-traffic-manager-webapp/azuredeploy.json

```diff
@@ -1,99 +1,93 @@
-            }
-        },
-        "uniqueDnsNameForWebApp": {
-            "type": "string",
-            "metadata": {
-                "description": "Relative DNS name for the WebApps, must be globally unique.  An index will be appended for each Web App."
-            }
-        }
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "uniqueDnsName": {
+      "type": "string",
+      "metadata": {
+        "description": "Relative DNS name for the traffic manager profile, resulting FQDN will be <uniqueDnsName>.trafficmanager.net, must be globally unique."
+      }
     },
+    "uniqueDnsNameForWebApp": {
+      "type": "string",
+      "metadata": {
+        "description": "Relative DNS name for the WebApps, must be globally unique.  An index will be appended for each Web App."
+      }
+    }
+  },
 
-    "variables": {
-        "tmApiVersion": "2015-11-01",
-        "apiVersion": "2015-08-01",
-        "webAppLocations": [ "West US", "East US" ],
-        "farmName": "default"
-    },
+  "variables": {
+    "tmApiVersion": "2015-11-01",
+    "apiVersion": "2015-08-01",
+    "webAppLocations": [ "West US", "East US" ],
+    "farmName": "default"
+  },
 
-    "resources": [
-        {
-            "apiVersion": "[variables('apiVersion')]",
-            "name": "[concat('DemoAppServicePlan', copyIndex())]",
-            "type": "Microsoft.Web/serverfarms",
-            "copy": {
-                "name": "farmloop",
-                "count": 2
-            },
-            "location": "[variables('webAppLocations')[copyIndex()]]",
-            "sku": {
-                "name": "S1",
-                "tier": "Standard"
-            }
-        },
-        {
-            "apiVersion": "[variables('apiVersion')]",
-            "name": "[concat(parameters('uniqueDnsNameForWebApp'), copyIndex())]",
-            "type": "Microsoft.Web/sites",
-            "copy": {
-                "name": "siteloop",
-                "count": 2
-            },
-            "location": "[variables('webAppLocations')[copyIndex()]]",
-            "dependsOn": [ "[concat('DemoAppServicePlan', copyIndex())]" ],
-            "properties": {
-                "serverFarmId": "[concat('DemoAppServicePlan', copyIndex())]"
-            }
+  "resources": [
+    {
+      "apiVersion": "[variables('apiVersion')]",
+      "name": "[concat('DemoAppServicePlan', copyIndex())]",
+      "type": "Microsoft.Web/serverfarms",
+      "copy": {
+        "name": "farmloop",
+        "count": "[length(variables('webAppLocations'))]"
+      },
+      "location": "[variables('webAppLocations')[copyIndex()]]",
+      "sku": {
+        "name": "S1",
+        "tier": "Standard"
+      }
+    },
+    {
+      "apiVersion": "[variables('apiVersion')]",
+      "name": "[concat(parameters('uniqueDnsNameForWebApp'), copyIndex())]",
```

Comment:
Is there are reason we cannot do this?  the sample uses geo location for demonstrating TM - would we expect developers to refer to resources by index or by location in this case (customer templates I've seen are more descriptive when labeling resources)